src/vm/arm/crthelpers.S

```diff
@@ -0,0 +1,156 @@
+        beq     AlignHalf
+        subs    r2, r2, #1
+        strb    r1, [r0], #1
+AlignHalf:
+        tst     r0, #2                                  // Check Half-word alignment
+        beq     BlockSet
+        subs    r2, r2, #2
+        strh    r1, [r0], #2
+        b       BlockSet
+
+LEAF_END_MARKED JIT_MemSet, _TEXT
+
+
+//EXTERN_C void __stdcall JIT_MemCpy(void* _dest, const void *_src, size_t count)
+LEAF_ENTRY JIT_MemCpy, _TEXT
+//
+// It only requires 4 byte alignment
+// and doesn't return a value
+
+        cmp     r2, #0                                  // quick check for 0 length
+        beq     ExitMemCpy                              // if zero, exit
+
+        tst     r0, #3                                  // skip directly to aligned if already aligned
+        beq     DestAligned                             // if 0, we're already aligned; go large
+        
+ByteLoop1:
+        subs    r2, r2, #1                              // decrement byte counter
+        ldrb    r3, [r1], #1                            // copy one byte
+        strb    r3, [r0], #1
+        beq     ExitMemCpy                              // if the byte counter hits 0, exit early
+        tst     r0, #3                                  // are we aligned now?
+        bne     ByteLoop1                               // nope, keep going
+
+DestAligned:
+        subs    r2, r2, #8                              // byte counter -= 8
+        blt     AlignedFinished                         // if that puts us negative, skip the big copy
+
+        tst     r1, #3                                  // is the 4-byte source aligned?
+        addne   r2, r2, #8                              // if not, fix the byte counter (+= 8)
+        bne     ByteLoop2                               // and do all the rest with bytes
+
+QwordLoop:    
+        subs    r2, r2, #8                              // decrement byte counter by 8
+        ldm     r1!, {r3,r12}                           // copy one qword
+        stm     r0!, {r3,r12}
+        bge     QwordLoop                               // loop until the byte counter goes negative
+        
+AlignedFinished:
+        adds    r2, r2, #4				// add 4 to recover a potential >= 4-byte tail
+        blt     AlignedFinished2
+        ldr     r3, [r1], #4
+        str     r3, [r0], #4
+        b       MaybeExitMemCpy
+AlignedFinished2:
+        adds    r2, r2, #4                              // add 4 more to the byte counter to recover
+
+MaybeExitMemCpy:
+        beq     ExitMemCpy                              // the remaining count
+        
+ByteLoop2:
+        subs    r2, r2, #1                              // decrement the counter
+        ldrb    r3, [r1], #1                            // copy one byte
+        strb    r3, [r0], #1
+        bne     ByteLoop2                               // loop until the counter hits 0
+
+ExitMemCpy:
+        bx      lr
+
+LEAF_END_MARKED JIT_MemCpy, _TEXT
+
```

Comment:
Implementation of JIT_MemCpy and JIT_MemSet for Unix ARM should mirror what we do for Unix AMD64: null check the pointer and tailcall into C runtime.